### PR TITLE
Cleanup - use modules defined in vmm-sys-util

### LINKED ISSUE #23: Remove sys_ioctls.rs from project
Once we have the functionality merged in the vmm-sys-util crate, we should remove the sys_ioctls.rs file and use the ioctl defined from there.

### PR DESCRIPTION
**Pull request by andreeaflorescu**
Changes:
- Removed ioctl macros from repository and use the ones defined in vmm-sys-util instead
- Fixed irqchip related tests
- use Eventfd instead of RawFd
- Remove Cargo.lock

Fixes: https://github.com/rust-vmm/kvm-ioctls/issues/23 and https://github.com/rust-vmm/kvm-ioctls/issues/39

### CODE REVIEW

**Review by jiangliu — APPROVED** [review]

**Review by acatangiu — COMMENTED** [review]
Why limit the interface to using `EventFd`? Why not use `& dyn AsRawFd`? Safety is guaranteed and the user gets more flexibility.

**Comment by jiangliu** [comment]
It's good to explicitly state an KVM EventFd is needed here.

**Comment by acatangiu** [comment]
> Why limit the interface to using EventFd? Why not use & dyn AsRawFd? Safety is guaranteed and the user gets more flexibility.

After going through the kernel KVM documentation and source, I see that it actually **requires** an `eventfd`. Disregard my previous comment.

File: src/ioctls/vm.rs [hunk 1]
```diff
@@ -530,7 +531,7 @@ impl VmFd {
     ///
     /// # Arguments
     ///
-    /// * `fd` - FD which will be signaled. When signaling, the usual `vmexit` to userspace
+    /// * `fd` - `EventFd` which will be signaled. When signaling, the usual `vmexit` to userspace
```
Comment (acatangiu) [inline src/ioctls/vm.rs:534]: nit: maybe use an explicit variable name
```suggestion
    /// * `evtfd` - `EventFd` which will be signaled. When signaling, the usual `vmexit` to userspace
```

**Review by acatangiu — APPROVED** [review]

**Review by alxiord — APPROVED** [review]

**Pull request merged**